File: 04-visitors/expressions/AddExpression.cpp
#include "AddExpression.h"

AddExpression::AddExpression(Expression *e1, Expression *e2): first(e1), second(e2) {}

int AddExpression::eval() const {
    return first->eval() + second->eval();
}

void AddExpression::Accept(Visitor* visitor) {
    visitor->Visit(this);
}

File: 2023-fall/nodes/expressions/AddExpression.py
from nodes.expressions.Expression import Expression


class AddExpression(Expression):

    def __init__(
            self,
            left: Expression,
            right: Expression
        ) -> None:
        self.left = left
        self.right = right
    
    def accept(self, visitor):
        return visitor.visit_add_expression(self)
